gnd/src/main/java/com/google/android/gnd/persistence/local/room/ProjectDao.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gnd.persistence.local.room;
+
+import androidx.room.Dao;
+import androidx.room.Delete;
+import androidx.room.Insert;
+import androidx.room.OnConflictStrategy;
+import androidx.room.Query;
+import io.reactivex.Completable;
+import io.reactivex.Maybe;
+import java.util.List;
+
+@Dao
+public interface ProjectDao {
+  @Insert(onConflict = OnConflictStrategy.REPLACE)
```

Comment:
I recently discovered that one replace the record is deleted then inserted, triggering cascading deletes of all related data! Please copy and paste this instead with a TODO to refactor once I merge the base class:

https://github.com/google/ground-android/pull/238/files#diff-1c91753f181f2553b29d35ecba14c670R30-R38